src/utils/utils.ts

```diff
@@ -4,8 +4,45 @@
 import { Stack } from 'aws-cdk-lib';
 import { Construct } from 'constructs';
 
-export const setTemplateDesc = (construct: Construct, sbtDesc: string) => {
+export const addTemplateTag = (construct: Construct, tag: string) => {
   const stackDesc = Stack.of(construct).templateOptions.description;
-  let description = stackDesc === undefined ? sbtDesc : stackDesc + ' - ' + sbtDesc;
+  const baseTelemetry = 'SaaS Builder Toolkit (uksb-1tupboc57)';
+  let description = stackDesc;
+  // There is no description, just make it telemetry + tags
+  if (stackDesc === undefined) {
+    description = appendTagToDescription(baseTelemetry, tag);
+  }
+  // There is a description, and it doesn't contain telemetry. We need to append telemetry + tags to it
+  else if (!stackDesc.includes(baseTelemetry)) {
+    description = appendTagToDescription(`${stackDesc} - ${baseTelemetry}`, tag);
+  }
+  // There is a telemetry description already
+  else {
+    description = appendTagToDescription(stackDesc, tag);
+  }
   Stack.of(construct).templateOptions.description = description;
 };
+
+const appendTagToDescription = (existingDescription: string, newTag: string): string => {
+  // Check if the existing description already has tags
+  if (existingDescription.includes('(Tags:')) {
+    // Extract the existing tags
+    const startIndex = existingDescription.indexOf('(Tags:') + 6;
```

Comment:
the `6` is throwing me off a little. Is it over the top to do something like this:

```
const tagOpeningString = '(Tags:';
const startIndex = existingDescription.indexOf(tagOpeningString) + tagOpeningString.length;
```